src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/SparseRangeInt32FrozenSet.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.Collections.Frozen
+{
+    /// <summary>Provides a frozen set to use when the value is an <see cref="int"/> and the default comparer is used and the items are in a contiguous range.</summary>
+    internal sealed class SparseRangeInt32FrozenSet : FrozenSetInternalBase<int, SparseRangeInt32FrozenSet.GSW>
+    {
+        private readonly int[] _items;
+        private readonly ulong[] _bits;
+        private readonly int _min;
+        private readonly int _max;
+
+        // assumes the items are in sorted order
+        internal SparseRangeInt32FrozenSet(int[] entries)
+            : base(EqualityComparer<int>.Default)
+        {
+            _items = entries;
+            _min = _items[0];
+            _max = _items[_items.Length - 1];
+
+            _bits = new ulong[((_max - _min + 1) + 63) / 64];
```

Comment:
```suggestion
            _bits = new ulong[((uint)(_max - _min + 1) + 63) / 64];
```
Yeah, an allocation + micro-optimization 😉
But it emits a bit-shift for the division with this change.

Reply:
Done.